library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -951,6 +1070,10 @@ private SampleQueue createSampleQueue(int id, int type) {
     sampleQueueTrackIds = Arrays.copyOf(sampleQueueTrackIds, trackCount + 1);
     sampleQueueTrackIds[trackCount] = id;
     sampleQueues = Util.nullSafeArrayAppend(sampleQueues, sampleQueue);
+    HlsMediaChunk mediaChunk = findChunkMatching(sourceChunk.uid);
+    if (mediaChunk != null) {
+      mediaChunk.setFirstSampleIndex(trackCount, 0);
```

Comment:
Maybe better to call sampleQueue.getWriteIndex() here as well to make the intention clear even though it's always 0 at this point.